### PR TITLE
Passing communications config as a k8s secret using existingSecret

### LINKED ISSUE #319: Load secret info from k8s Secret objects.
**Is your feature request related to a problem? Please describe.**

I'm using HelmResource CRD objects to install and manage helm v3 packages.  Putting a secret in clear text in a HelmResource yaml file, to be committed in a git repository, is a no-no.

Secrets need to be stored separately, ideally in a sealed secret.

**Describe the solution you'd like**

Instead of storing secret information in the `values.yaml`, the `values.yaml` should allow pointing at an exist k8s Secret object by name and namespace.

**Describe alternatives you've considered**

There isn't really any way to use the chart with the current design without leaking secrets.

**Additional context**

I like the project. It looks cool!

### PR DESCRIPTION
**Pull request by ebrianne**
##### ISSUE TYPE
 - Feature Pull Request

##### SUMMARY
I have implemented in the helm chart the possibility to pass the communication config as a k8s secret and configure in the value files via an existingSecret. It can be configured in the values file via these:

```yaml
communications:
    existingSecret: false
    existingSecretName: ""
```

Fixes #319 #211 #226 #233

### CODE REVIEW

File: helm/botkube/templates/_helpers.tpl [hunk 1]
```diff
@@ -41,3 +41,7 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{- define "botkube.CommunicationsSecretName" -}}
+{{- default (include "botkube.fullname" .) (.Values.communications.existingSecretName) -}}
```
Comment (bhavin192) [inline helm/botkube/templates/_helpers.tpl:46]: I think the default name should be kept what it was before i.e. `{{ include "botkube.fullname" . }}-communication-secret`. That way the name will indicate correctly that it is a communication related secret. Also, we won't break anything if someone is depending on that name to do something (less likely, but still).

File: helm/botkube/values.yaml [hunk 2]
```diff
@@ -273,6 +273,11 @@ config:
 
 # Communication settings
 communications:
+
+  # Communication secret already existing
```
Comment (bhavin192) [inline helm/botkube/values.yaml:277]:
```suggestion
  # Using existing Communication secret
```

**Review by bhavin192 — CHANGES_REQUESTED** [review]
Thank you for creating the change, this looks good to me. I have suggested a change related to name of the secret.

**Comment by pierluigilenoci** [comment]
@ebrianne could you please implement the suggestion in your PR?

**Comment by ebrianne** [comment]
I will implemented the changes as soon as I have time... today won't be possible. I ll try over the week end.

**Comment by pierluigilenoci** [comment]
@ebrianne any news?

**Comment by ebrianne** [comment]
I have just pushed the suggested changes

**Comment by pierluigilenoci** [comment]
@bhavin192 could you please take a look?

**Comment by pierluigilenoci** [comment]
@PrasadG193 could you please take a look?

File: helm/botkube/templates/communicationsecret.yaml [hunk 3]
```diff
@@ -1,7 +1,8 @@
+{{- if not .Values.communications.existingSecret -}}
```
Comment (bhavin192) [inline helm/botkube/templates/communicationsecret.yaml:1]:
```suggestion
{{- if not .Values.communications.existingSecretName -}}
```

How about this?
Right now, if one does following, current logic tries to use the secret even if we have set it to not use it (`false`).
```
 --set communications.existingSecretName="test" --set communications.existingSecret=false
```
With the suggested change we can remove the `existingSecret` value as well.

**Review by bhavin192 — CHANGES_REQUESTED** [review]

**Comment by pierluigilenoci** [comment]
@ebrianne now it's your turn 😜

File: helm/botkube/values.yaml [hunk 4]
```diff
@@ -273,6 +273,11 @@ config:
 
 # Communication settings
 communications:
+
+  # Using existing Communication secret
+  existingSecret: false
+  existingSecretName: ""
```
Comment (bhavin192) [inline helm/botkube/values.yaml:279]:
```suggestion
  # Using existing Communication secret
  existingSecretName: ""
```

**Comment by pierluigilenoci** [comment]
i8O0J31
😆

**Review by bhavin192 — APPROVED** [review]
:tada: :rocket:

**Comment by onedr0p** [comment]
Please push a new release of the chart this is very much needed 🙏🏼

**Comment by rhummelmose** [comment]
Why isn't a Helm chart with this released yet?

**Comment by pierluigilenoci** [comment]
@rhummelmose nope. 😢

**Comment by pierluigilenoci** [comment]
@bhavin192 @PrasadG193 can you help to make a release?

**Comment by PrasadG193** [comment]
Hey @pierluigilenoci, I am already working on it. We will make a release this week. Sorry for the delay.

**Comment by rhummelmose** [comment]
Thanks, really need this (y)

**Comment by onedr0p** [comment]
@PrasadG193 could you please also make the GHCR image the default in the chart <3

https://github.com/infracloudio/botkube/pull/498

**Comment by rhummelmose** [comment]
@PrasadG193 Is this coming? :)

**Comment by onedr0p** [comment]
What's holding up the release, do you all need help on anything?

**Pull request merged**